src/DotVVM.Framework.Tests.Common/Binding/BindingCompilationTests.cs

```diff
@@ -130,6 +130,64 @@ public void BindingCompiler_Valid_StringLiteralInSingleQuotes()
             Assert.AreEqual(ExecuteBinding("StringProp + 'def'", viewModel), "abcdef");
         }
 
+        [TestMethod]
+        [DataRow(@"$'Non-Interpolated'", "Non-Interpolated")]
+        [DataRow(@"$'Non-Interpolated {{'", "Non-Interpolated {")]
+        [DataRow(@"$'Non-Interpolated {{ no-expression }}'", "Non-Interpolated { no-expression }")]
+        public void BindingCompiler_Valid_InterpolatedString_NoExpressions(string expression, string evaluated)
+        {
+            var binding = ExecuteBinding(expression);
+            Assert.AreEqual(evaluated, binding);
+        }
+
+        [TestMethod]
+        [DataRow(@"$'} Malformed'", "Could not find matching opening character '{' for an interpolated expression")]
+        [DataRow(@"$'{ Malformed'", "Could not find matching closing character '}' for an interpolated expression")]
+        [DataRow(@"$'Malformed {expr'", "Could not find matching closing character '}' for an interpolated expression")]
+        [DataRow(@"$'Malformed expr}'", "Could not find matching opening character '{' for an interpolated expression")]
+        [DataRow(@"$'Malformed {'", "Could not find matching closing character '}' for an interpolated expression")]
+        [DataRow(@"$'Malformed }'", "Could not find matching opening character '{' for an interpolated expression")]
+        [DataRow(@"$'Malformed {}'", "Expected expression, but instead found empty")]
+        public void BindingCompiler_Valid_InterpolatedString_Malformed(string expression, string errorMessage)
+        {
+            try
+            {
+                ExecuteBinding(expression);
+            }
+            catch (Exception e)
+            {
+                // Get inner-most exception
+                var current = e;
+                while (current.InnerException != null)
+                    current = current.InnerException;
+
+                Assert.AreEqual(typeof(BindingCompilationException), current.GetType());
+                StringAssert.Contains(current.Message, errorMessage);
+            }
+        }
+
+        [TestMethod]
+        [DataRow(@"$""Interpolated {StringProp} {StringProp}""", "Interpolated abc abc")]
+        [DataRow(@"$'Interpolated {StringProp} {StringProp}'", "Interpolated abc abc")]
+        [DataRow(@"$'Interpolated {StringProp.Length}'", "Interpolated 3")]
+        public void BindingCompiler_Valid_InterpolatedString_WithSimpleExpressions(string expression, string evaluated)
```

Comment:
Could you please also test that nested string interpolations work? For example `$"{ string.Join(", ", MyArray) }"`, I think it may be broken now.